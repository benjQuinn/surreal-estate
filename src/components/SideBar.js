import React from "react";
import "../styles/side-bar.css";
import { Link } from "react-router-dom";
import { BsFilterCircleFill } from "react-icons/bs";

const SideBar = () => {
  return (
    <div className="side-bar">
      <BsFilterCircleFill className="side-bar-icon" />
      <ul className="side-bar-links">
        <li className="side-bar-links-item">
          <Link
            to={`/?query={"city": "Manchester"}`}
            className="item"
            data-testid="side-bar-manchester-link"
          >
            Manchester
          </Link>
        </li>
        <li className="side-bar-links-item">
          <Link to={`/?query={"city": "Leeds"}`} className="item">
            Leeds
          </Link>
        </li>
        <li className="side-bar-links-item">
          <Link to={`/?query={"city": "Liverpool"}`} className="item">
            Liverpool
          </Link>
        </li>
        <li className="side-bar-links-item">
          <Link to={`/?query={"city": "Sheffield"}`} className="item">
            Sheffield
          </Link>
        </li>
      </ul>
    </div>
  );
};

export default SideBar;
