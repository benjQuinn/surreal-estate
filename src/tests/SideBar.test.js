import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import SideBar from "../components/SideBar";

describe("SideBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Router>
        <SideBar />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders links correctly", () => {
    const { getByText } = render(
      <Router>
        <SideBar />
      </Router>
    );
    const manchesterLink = getByText(/manchester/i);
    const leedsLink = getByText(/leeds/i);
    const liverpoolLink = getByText(/liverpool/i);
    const sheffieldLink = getByText(/sheffield/i);

    expect(manchesterLink).toBeInTheDocument();
    expect(leedsLink).toBeInTheDocument();
    expect(liverpoolLink).toBeInTheDocument();
    expect(sheffieldLink).toBeInTheDocument();
  });
});
