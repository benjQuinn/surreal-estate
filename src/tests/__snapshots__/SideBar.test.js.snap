// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="side-bar"
  >
    <svg
      class="side-bar-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zM3.5 5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1 0-1zM5 8.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm2 3a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 0 1h-1a.5.5 0 0 1-.5-.5z"
      />
    </svg>
    <ul
      class="side-bar-links"
    >
      <li
        class="side-bar-links-item"
      >
        <a
          class="item"
          data-testid="side-bar-manchester-link"
          href="/?query={\\"city\\": \\"Manchester\\"}"
        >
          Manchester
        </a>
      </li>
      <li
        class="side-bar-links-item"
      >
        <a
          class="item"
          href="/?query={\\"city\\": \\"Leeds\\"}"
        >
          Leeds
        </a>
      </li>
      <li
        class="side-bar-links-item"
      >
        <a
          class="item"
          href="/?query={\\"city\\": \\"Liverpool\\"}"
        >
          Liverpool
        </a>
      </li>
      <li
        class="side-bar-links-item"
      >
        <a
          class="item"
          href="/?query={\\"city\\": \\"Sheffield\\"}"
        >
          Sheffield
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
